src/TLuaInterpreter.cpp

```diff
@@ -3073,65 +3073,95 @@ int TLuaInterpreter::getServerEncodingsList(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getOS
 int TLuaInterpreter::getOS(lua_State* L)
 {
+    auto pushProcessor = [&] () {
+#if defined(Q_PROCESSOR_IA64)
+        lua_pushstring(L, "ia64");
+        return 1;
+#elif defined(Q_PROCESSOR_X86_64)
+        lua_pushstring(L, "x86 (64-bit)");
+        return 1;
+#elif defined(Q_PROCESSOR_X86_32)
+        lua_pushstring(L, "x86 (32-bit)");
+        return 1;
+#elif defined(Q_PROCESSOR_POWER_64)
+        lua_pushstring(L, "ppc (64-bit)");
+        return 1;
+#elif defined(Q_PROCESSOR_POWER_32)
+        lua_pushstring(L, "ppc (32-bit)");
+        return 1;
+#elif defined(Q_PROCESSOR_ARM_V7)
+        lua_pushstring(L, "arm7");
+        return 1;
+#elif defined(Q_PROCESSOR_ARM_V6)
+        lua_pushstring(L, "arm6");
+        return 1;
+#elif defined(Q_PROCESSOR_ARM_V5)
+        lua_pushstring(L, "arm5");
+        return 1;
```

Comment:
I don't think this level of detail will help, lets go with a single `Q_PROCESSOR_ARM` for now until we have a need to differentiate it more.

Reply:
I think it would be advisable to keep the distinctions IIRC different Raspberry Pis have difference types of ARM processors and when you consider the IOT/SBC side of things I think we'd be covering our arses to report the differences if we can detect them...